src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -73,7 +87,11 @@ class CoreModuleImpl @Inject() (
     marathonSchedulerDriverHolder,
 
     // internal core dependencies
-    offerMatcherManagerModule.globalOfferMatcher)
+    StopOnFirstMatchingOfferMatcher(
+      offerMatcherReconcilerModule.offerMatcherReconciler,
+      offerMatcherManagerModule.globalOfferMatcher
```

Comment:
Oh, that's basically nesting. So if the reconciler matches, take that result. Otherwise just proceed to the globalMatcher as before. The internal logic of the globalOfferMatcher is not effected -- it encapsulates all the other matchers.